realsense2_camera/include/ros_sensor.h

```diff
@@ -32,9 +32,9 @@ namespace realsense2_camera
     class FrequencyDiagnostics
     {
     public:
-    FrequencyDiagnostics(std::string name, int expected_frequency, std::shared_ptr<diagnostic_updater::Updater> updater):
+    FrequencyDiagnostics(std::string name, int expected_frequency, double min_fps_threshold, std::shared_ptr<diagnostic_updater::Updater> updater):
             _name(name),
-            _min_freq(expected_frequency), _max_freq(expected_frequency),
+            _min_freq(min_fps_threshold), _max_freq(expected_frequency),
```

Comment:
where does this expected frequency come from? it is currently 90% of the actual frequency. To make it more generic, you could add a parameter to set this e.g. to 80 or 70%

Reply:
I will check. Looks like `expected_frequency` is` color_fps: 30.` in .yaml file

Reply:
![grafik](https://github.com/user-attachments/assets/cd39e530-b5e5-4338-97b7-c17a17b917a5)

there is min=27

Reply:
> where does this expected frequency come from? it is currently 90% of the actual frequency. To make it more generic, you could add a parameter to set this e.g. to 80 or 70%

I can add parameter and we will set this value from yaml file

Reply:
see
http://docs.ros.org/en/jade/api/diagnostic_updater/html/structdiagnostic__updater_1_1FrequencyStatusParam.html

-> the 0.1 is the tolerance

So, I think best solution here is to add a parameter, so that we can specify this tolerance,e .g. `color_freq_tolerance: 0.1`

Reply:
Yes, now it works this way:

![image](https://github.com/user-attachments/assets/9c1fe893-d627-4a86-826f-a3b1274dbf15)

Set `color_freq_tolerance: 0.5` to accept anything above 15fps.